src/ResourceManager.cpp

```diff
@@ -0,0 +1,151 @@
+}
+
+std::shared_ptr<Shader> ResourceManager::load_shader(std::string const& compute_path)
+{
+    std::string const& key = compute_path; // No need to even call generate_key()
+    auto resource_ptr = get_from_vector<ShaderDX11>(key);
+
+    if (resource_ptr != nullptr)
+        return resource_ptr;
+
+    resource_ptr = std::dynamic_pointer_cast<ShaderDX11>(ShaderFactory::create(compute_path));
+    m_shaders.emplace_back(resource_ptr);
+    names_to_shaders.insert(std::make_pair(key, m_shaders.size() - 1));
+
+    return resource_ptr;
+}
+
+std::shared_ptr<Shader> ResourceManager::load_shader(std::string const& vertex_path, std::string const& fragment_path)
+{
+    std::stringstream stream;
+    stream << vertex_path << fragment_path;
+    std::string const& key = generate_key(stream);
+
+    auto resource_ptr = get_from_vector<ShaderDX11>(key);
+
+    if (resource_ptr != nullptr)
+        return resource_ptr;
+
+    resource_ptr = std::dynamic_pointer_cast<ShaderDX11>(ShaderFactory::create(vertex_path, fragment_path));
+    m_shaders.emplace_back(resource_ptr);
+    names_to_shaders.insert(std::make_pair(key, m_shaders.size() - 1));
+
+    return resource_ptr;
+}
+
+std::shared_ptr<Shader> ResourceManager::load_shader(std::string const& vertex_path, std::string const& fragment_path,
+    std::string const& geometry_path)
+{
+    std::stringstream stream;
+    stream << vertex_path << fragment_path << geometry_path;
+    std::string const& key = generate_key(stream);
+
+    auto resource_ptr = get_from_vector<ShaderDX11>(key);
+
+    if (resource_ptr != nullptr)
+        return resource_ptr;
+
+    resource_ptr = std::dynamic_pointer_cast<ShaderDX11>(ShaderFactory::create(vertex_path, fragment_path, geometry_path));
+    m_shaders.emplace_back(resource_ptr);
+    names_to_shaders.insert(std::make_pair(key, m_shaders.size() - 1));
+
+    return resource_ptr;
+}
+
+std::shared_ptr<Shader> ResourceManager::load_shader(std::string const& vertex_path,
+    std::string const& tessellation_control_path, std::string const& tessellation_evaluation_path,
+    std::string const& fragment_path)
+{
+    std::stringstream stream;
+    stream << vertex_path << tessellation_control_path << tessellation_evaluation_path << fragment_path;
+    std::string const& key = generate_key(stream);
+
+    auto resource_ptr = get_from_vector<ShaderDX11>(key);
+
+    if (resource_ptr != nullptr)
+        return resource_ptr;
+
+    resource_ptr = std::dynamic_pointer_cast<ShaderDX11>(ShaderFactory::create(vertex_path, tessellation_control_path, tessellation_evaluation_path, fragment_path));
+    m_shaders.emplace_back(resource_ptr);
+    names_to_shaders.insert(std::make_pair(key, m_shaders.size() - 1));
+
+    return resource_ptr;
+}
+
+std::shared_ptr<Model> ResourceManager::load_model(std::string const& path_to_model,
+    std::shared_ptr<Material> const& material)
+{
+    // This is only for consistency.
+    return Model::create(path_to_model, material);
```

Comment:
Since Model is a component this is actually not consistent. Moreover, models will be created in the editor using the `Model::create();` method with empty parameters anyway.